sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/PTransformTranslation.java

```diff
@@ -522,11 +521,24 @@ public RunnerApi.PTransform translate(
         }
 
         if (spec.getUrn().equals(BeamUrns.getUrn(SCHEMA_TRANSFORM))) {
+          ExternalTransforms.SchemaTransformPayload payload =
+              ExternalTransforms.SchemaTransformPayload.parseFrom(spec.getPayload());
+          String identifier = payload.getIdentifier();
           transformBuilder.putAnnotations(
-              SCHEMATRANSFORM_URN_KEY,
-              ByteString.copyFromUtf8(
-                  ExternalTransforms.SchemaTransformPayload.parseFrom(spec.getPayload())
-                      .getIdentifier()));
+              BeamUrns.getConstant(Annotations.Enum.SCHEMATRANSFORM_URN_KEY),
+              ByteString.copyFromUtf8(identifier));
+          if (identifier.equals(MANAGED_TRANSFORM_URN)) {
+            Schema configSchema =
+                SchemaTranslation.schemaFromProto(payload.getConfigurationSchema());
+            Row configRow =
+                RowCoder.of(configSchema).decode(payload.getConfigurationRow().newInput());
+            String underlyingIdentifier =
+                MoreObjects.firstNonNull(
+                    configRow.getString("transform_identifier"), "unknown_identifier");
+            transformBuilder.putAnnotations(
```

Comment:
Is there a valid case where "transform_identifier" would not be set ? If not we should just error out.